### PR TITLE
fix(curriculum): Improved the tests for Find the length of a String challenge

### PR DESCRIPTION
**Pull request by RandellDawson**
- [x] I have read [freeCodeCamp's contribution guidelines](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/CONTRIBUTING.md).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `master` branch of freeCodeCamp.
- [x] None of my changes are plagiarized from another source without proper attribution.
- [x] All the files I changed are in the same world language (for example: only English changes, or only Chinese changes, etc.)
- [x] My changes do not use shortened URLs or affiliate links.

I have noticed a few campers get confused regarding the test messages for this challenge.  The reason for the confusion is if the user changes the original setup code, but the code would still technically result in the correct solution, the test message would fail but the reason given would not match the reason stated because the test combines the following checks but displays "You should be getting the length of lastName by using .length like this: lastName.length." if either fail.
1. Check that the original setup code is not changed
2. Check the the user gets the length of lastName by using .length like this: lastName.length

See https://www.freecodecamp.org/forum/t/help-w-js-challenge-suspected-bug/274365 for the full discussion.

This PR creates an additional test to make sure the user does not change the original setup code which the tests use.

### CODE REVIEW

File: curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/find-the-length-of-a-string.english.md [hunk 1]
```diff
@@ -22,10 +22,12 @@ Use the <code>.length</code> property to count the number of characters in the <
 
 ```yml
 tests:
+  - text: 'You should not change the variable declarations in the <code>// Setup</code> section.'
+    testString: assert(code.match(/var lastNameLength = 0;/) && code.match(/var lastName = "Lovelace";/));   
   - text: <code>lastNameLength</code> should be equal to eight.
-    testString: assert((function(){if(typeof lastNameLength !== "undefined" && typeof lastNameLength === "number" && lastNameLength === 8){return true;}else{return false;}})(), '<code>lastNameLength</code> should be equal to eight.');
+    testString: assert(typeof lastNameLength !== 'undefined' && typeof lastNameLength === 'number' && lastNameLength === 8);
```
Comment (ValeraS) [inline curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/find-the-length-of-a-string.english.md]:
```suggestion
    testString: assert(typeof lastNameLength !== 'undefined' && lastNameLength === 8); 
```

**Review by ValeraS — CHANGES_REQUESTED** [review]
we can delete an after test section.

**Comment by RandellDawson** [comment]
@ValeraS So we do not need the following for this challenge? That was already there.
````

### After Test
<div id='js-teardown'>
```js
if(typeof lastNameLength !== "undefined"){(function(){return lastNameLength;})();}
```
</div>

````

**Comment by ValeraS** [comment]
I don't see why it needed.

**Review by ValeraS — APPROVED** [review]

**Review by moT01 — APPROVED** [review]
LGTM! Thanks @RandellDawson! There was no issue for this I take it?

**Pull request merged**